compiler/ccgexprs.nim

```diff
@@ -1879,9 +1879,13 @@ proc genArrayLen(p: BProc, e: PNode, d: var TLoc, op: TMagic) =
     # Bug #9279, len(toOpenArray()) has to work:
     if a.kind in nkCallKinds and a[0].kind == nkSym and a[0].sym.magic == mSlice:
       # magic: pass slice to openArray:
+      var m: TLoc
```

Comment:
there's a `genSlice` that potentially could be used, but from the looks of it, it also lacks bounds checks

Reply:
I don't know. In this case, there is no need to generate slice. A bound check should suffice.

Reply:
so I've noticed, ie the cgen has special slice handling all over, often duplicated, and each place is slightly different leading to bugs like this - collecting it all in one place seems like a worthwhile effort - basically, treating everything as a reified openarray would significantly simplify the flow and thus make it less error-prone.